src/api/public/apidocs-new/OBS-v2.10.50.yaml

```diff
@@ -8,7 +8,18 @@ tags:
   - name: Groups
 
 info:
-  description: Open Build Service API
+  description: |
+    The _Open Build Service API_ is a XML API.
+    This documentation is based on the _OpenAPI 3.0.0_ specification and powered by _Swagger_.
+
+    To authenticate, use [HTTP basic authentication](https://en.wikipedia.org/wiki/Basic_access_authentication) by passing the _Authorization_ header in the form of `Authorization: Basic <credentials>`.
+
+    We don't have any versioning.
```

Comment:
sure we have.

Or do you really speak about some complete new api I never heard about?

Reply:
I'm writing about the existing API. Could you point me to the versioning that we have? Because I don't see any implementation of that.

Reply:
check for the /source/ routes, you can specify the revision via the "rev" CGI parameter. This works for everything which is considered to be sources, this includes also meta data like project and package _meta files or attributes.

Reply:
I am referring to API versioning. The only version of the API which is available, is the latest one. This is why I'm writing that we don't have any versioning. To be extra clear, I'll add _API_ in front of _versioning_.

What you are referring to, in the _/sources/_ routes, is about requesting a specific version of a source.

Reply:
@dmarcoux check this https://github.com/openSUSE/open-build-service/blob/master/src/api/app/controllers/application_controller.rb#L23

```
 def add_api_version
    response.headers['X-Opensuse-APIVersion'] = (CONFIG['version']).to_s
  end
```

Reply:
for me it looks like a version. We add it to every API request...

Reply:
I did see that. But it's not API versioning. It's telling clients what the version of the API, which is always referring to the latest version deployed in production.

If we had API versioning, it could be something like below. It's just an example:
- https://build.opensuse.org/v1/about (old version which isn't changed, except on rare occasions like fixing a security vulnerability)
- https://build.opensuse.org/v2/about (current version which introduced breaking changes from v1, not actively developed)
- https://build.opensuse.org/v3/about (new version which will introduce breaking changes from v2, actively developed and not released)

Reply:
k, we do kind of "versioning" by introducing alternative routes where needed. But this is only needed in rare cases.

Anyway, I would not mention something on the frontpage what is not there. It is imho way to prominent there and we should focus on what is there. Maybe giving a highlevel overview what are the /source /build or /attribute areas for example.

Reply:
API versioning is quite common nowadays and it's good to be clear that our API doesn't have this. As for the documentation specific to endpoints like _/source_ and others, we're going to tackle that in the future. It's going to be like the endpoints which are already documented with Swagger.